lib/json_api_controller/precondition_check.rb

```diff
@@ -0,0 +1,28 @@
+module JsonApiController
+  module PreconditionCheck
+    def precondition_check
+      if !precondition
+        raise PreconditionNotPresent, precondition_error_msg
+      elsif precondition_fails?
+        raise PreconditionFailed
+      end
+    end
+    
+    def precondition
+      case action_name
+      when "show"
+        request.headers["If-Modified-Since"]
+      else
+        request.headers["If-Unmodified-Since"]
```

Comment:
Is the header being used consistent as per [here](https://github.com/zooniverse/Panoptes/issues/296#issuecomment-69229832)? Do we need to update the issue to reflect which header should be set client side?